src/odemis/driver/smaract.py

```diff
@@ -1541,9 +1548,16 @@ def _doReference(self, future):
         finally:
             # We only notify after updating the position so that when a listener
             # receives updates both values are already updated.
-            if self._is_referenced():
+            # only move if no exception was raised.
+            if self._is_referenced() and not raised:
                 self.referenced._value = {a: True for a in self.axes.keys()}
                 self._updatePosition()
+                if self._deactive_pos_after_ref is not None:
+                    logging.info("Moving axes to deactivated position %s after referencing", self._deactive_pos_after_ref)
+
+                    self._checkMoveAbs(self._deactive_pos_after_ref)
+                    f = self._createMoveFuture()
```

Comment:
BTW, could you remove the "ref" argument from `_createMoveFuture()`, as it's unused.